frail/verilog_printer.py

```diff
@@ -298,7 +301,9 @@ def mod_str_from_ports(io_ports: Dict[int, List[ModulePort]],
             if tab_str + io_str not in top_module_io:
                 top_module_io.append(tab_str + io_str)
             inter_str = port.name
-        module_inst_str += tab_str + tab_str + f".{port.name}({inter_str}),\n"
+        module_port_str = tab_str + tab_str + f".{port.name}({inter_str}),\n"
+        if module_port_str not in module_inst_str:
```

Comment:
Or was this were a port would appear multiple times. I could see that multipel modules use the same variable.
Could you clarifiy if here or above was the issue.